packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1576,10 +1580,11 @@ Future<T> showGeneralDialog<T>({
   Color barrierColor,
   Duration transitionDuration,
   RouteTransitionsBuilder transitionBuilder,
+  NavigatorState navigator,
```

Comment:
Looks like context doesn't have to be required anymore on this method. But we should have an assert that either context or navigator is not null.